### ISSUE TITLE
1Inch PriceEstimate and amount is inverse. 

### ISSUE THREAD

**Issue by fleupold** [gnosis/dex-liquidity-provision#515]
The console log says:

```
1inch price: 353.5065894727491 USDC bought for 1 WETH
```

But what we estimate is https://api.1inch.exchange/v1.1/quote?fromTokenSymbol=USDC&toTokenSymbol=WETH&amount=1000000

which gives us the amount WETH bought for 1 USDC.

At the time of writing this price ($353) is significantly different than the inverse ($335)

**Comment by bh2smith** [comment]
I see from this log that 1 WETH would have cost 353 USDC which seems pretty accurate for the date and time that this post was published. I am not sure where you are getting 335 from here? 

If I understand the issue correctly, the problem here is simply that the log itself is inverted and should say

```
1inch price: 1 WETH bought for 353.5065894727491 USDC
```

**Comment by bh2smith** [comment]
Ok, so after a little digging (again) into the definition of Base and Quote when mapped to 1Inch terms "to" and "from" token found [here](https://www.axiory.com/trading-resources/trading-terms/currency-quotes)

I have extracted the following relevant paragraph

```
For example, in a EUR/JPY currency pair, euro is the base currency while the Japanese yen is the quote currency. And when someone mentions the price of the currency pair, it means the amount of the second currency necessary to buy the first one. In this case, the price of a EUR/JPY pair is the amount of yen needed to buy one euro.
```

Meaning here where
```
fromToken = quoteToken = USDC
toToken = baseToken = WETH
```
The log statement should read

"needed to buy 1" in place of "bought for 1" in lines 184 and 185

https://github.com/gnosis/dex-liquidity-provision/blob/aab35854c1c9e6814c6d81e6bd7063ac5570545c/scripts/utils/price_utils.js#L184-L187

**Comment by fleupold** [comment]
> I am not sure where you are getting 335 from here?

Price at the time (Oct 8, 11:00 CET) was around $335 (cf. https://www.tradingview.com/symbols/ETHUSD/)

<img width="506" alt="Screen Shot 2020-10-13 at 11 59 17" src="https://user-images.githubusercontent.com/1200333/95846303-921cc780-0d4b-11eb-8f2f-55816410fe2c.png">

**Comment by bh2smith** [comment]
Oh wow! This is strange. Then perhaps there is something off about 1Inch as a price estimation service. Given that 353 and 335 are so different but clearly not multiplicative inverses of each other I think we can assume that the to and from token mapping is correct.

**Comment by fleupold** [comment]
I think the spread was just very large ($353 vs $316 iirc), which might have something to do with the fact that 1Inch seemed to route the price estimate via mooniswap (where liquidity might not be great).

**Comment by bh2smith** [comment]
In any case, I will look into making a reasonable test for this, but I think the issue itself regarding the inverted log is solved by updating the log statement as in #534

**Comment by bh2smith** [comment]
I am currently running this price estimation with `WETH/USDC` and getting 385.9 while other price sources are showing 383.2 which is also still off by a bit. Perhaps we could make a test that queries two sources and attempts to ensure that they agree within 2% of each other.... Although this would involve having a test that makes 2 http requests and would not be deterministic.
